functions/src/lobby.ts

```diff
@@ -0,0 +1,56 @@
+import * as functions from "firebase-functions";
+import { lobbyCollection, userCollection } from "./firestore-collections";
+import { Lobby } from "./firestore-types/lobby";
+import { UserData } from "./firestore-types/users";
+
+export const startGame = functions.https.onCall(async (data: { code: string }, context) => {
+  // no auth then you shouldn't be here
+  if (context.auth === undefined) {
+    return { error: "Not Signed In" };
+  }
+  // get lobby doc
+  const lobby = await lobbyCollection.doc(data.code).get();
+  if (lobby.exists === false) {
+    return { error: "Lobby doesn't exist!" };
+  }
+  // check if the request is coming from the host of the game
+  const { uids } = lobby.data() as Lobby;
+  if (context.auth.uid !== uids[0]) {
+    return { error: "Not allowed to start game!" };
+  }
+
+  return lobbyCollection.doc(data.code).update({ state: "PROMPT" });
+});
+
+export const addPlayer = functions.https.onCall(async (data: { code: string }, context) => {
```

Comment:
we should probobly makes types for the imnput data and share the types between the server and the client so that if we want to change it in the future it will force us to change it in both places